pytorch_lightning/loops/fit_loop.py

```diff
@@ -0,0 +1,254 @@
+        self.training_loop.batch_loop.skip_backward = value
+
+    @property
+    def done(self) -> bool:
+        # TODO(@awaelchli): Move track steps inside training loop and move part of these condition inside training loop
+        stop_steps = self.max_steps is not None and self.global_step >= self.max_steps
+        stop_epochs = self.max_epochs is not None and self.current_epoch >= self.max_epochs
+
+        should_stop = False
+        if self.trainer.should_stop:
+            # early stopping
+            met_min_epochs = self.current_epoch >= self.min_epochs if self.min_epochs else True
+            met_min_steps = self.global_step >= self.min_steps if self.min_steps else True
+            if met_min_epochs and met_min_steps:
+                should_stop = True
+            else:
+                log.info(
+                    'Trainer was signaled to stop but required minimum epochs'
+                    f' ({self.min_epochs}) or minimum steps ({self.min_steps}) has'
+                    ' not been met. Training will continue...'
+                )
+                self.trainer.should_stop = False
+
+        return stop_steps or should_stop or stop_epochs
+
+    def connect(self, trainer: 'pl.Trainer', *args, **kwargs):
+        self.trainer = trainer
+        self.training_loop.connect(trainer)
+
+    def reset(self) -> None:
+        self.iteration_count = 0
+
+    def run(self):
+        if not self._should_skip_training():
+            return super().run()
+
+    def on_run_start(self):
+        self.results.to(device=self.trainer.lightning_module.device)
+        self.trainer.call_hook("on_train_start")
+
+    def on_advance_start(self):
+        model = self.trainer.lightning_module
+
+        # reset train dataloader
+        if self.current_epoch != 0 and self.trainer.reload_dataloaders_every_epoch:
+            self.trainer.reset_train_dataloader(model)
+
+        # TODO: specify the possible exception
+        with suppress(Exception):
+            # set seed for distributed sampler (enables shuffling for each epoch)
+            self.trainer.train_dataloader.sampler.set_epoch(self.current_epoch)
+
+        # changing gradient according accumulation_scheduler
+        self.trainer.accumulation_scheduler.on_train_epoch_start(self.trainer, self.trainer.lightning_module)
+
+        # stores accumulated grad fractions per batch
+        self.training_loop.batch_loop.accumulated_loss = TensorRunningAccum(
+            window_length=self.trainer.accumulate_grad_batches
+        )
+
+        # hook
+        self.trainer.logger_connector.on_epoch_start()
+        self.trainer.call_hook("on_epoch_start")
+        self.trainer.call_hook("on_train_epoch_start")
+
+    def advance(self):
+        train_dataloader = self.trainer.accelerator.process_dataloader(self.trainer.train_dataloader)
+        train_dataloader = self.trainer.data_connector.get_profiled_train_dataloader(train_dataloader)
```

Comment:
- should the train dataloader be owned by the loop or by the trainer? 
- is there a risk that self.trainer.train_dataloader falls out of sync with the train_dataloader used here after the accelerator processing and profiling?